helm-charts/common/nginx/gateway-values.yaml

```diff
@@ -0,0 +1,243 @@
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_connect_timeout 6000;
+            proxy_send_timeout 6000;
+            proxy_read_timeout 6000;
+            send_timeout 6000;
+        }
+
+        # Chathistory service
+        location /v1/chathistory {
+            proxy_pass http://${CHATHISTORY_SERVICE_IP}:${CHATHISTORY_SERVICE_PORT}/v1/chathistory;
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_buffering off;
+            proxy_cache off;
+            proxy_request_buffering off;
+            gzip off;
+        }
+
+        # Prompt service
+        location /v1/prompt {
+            proxy_pass http://${PROMPT_SERVICE_IP}:${PROMPT_SERVICE_PORT}/v1/prompt;
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_buffering off;
+            proxy_cache off;
+            proxy_request_buffering off;
+            gzip off;
+        }
+
+        # Keycloak service
+        location /auth {
+            proxy_pass http://keycloak.keycloak.svc.cluster.local:80;
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_buffering off;
+            proxy_cache off;
+            proxy_request_buffering off;
+            gzip off;
+        }
+
+        # Error pages
+        error_page 404 /404.html;
+        location = /404.html {
+            root /usr/share/nginx/html;
+        }
+
+        error_page 500 502 503 504 /50x.html;
+        location = /50x.html {
+            root /usr/share/nginx/html;
+        }
+    }
+
+# Optional: Ingress configuration for gateway mode
+ingress:
+  enabled: false
+  className: ""
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+# Health check probes for gateway mode
+readinessProbe:
+  httpGet:
+    path: /v1/health_check
+    port: nginx
+  initialDelaySeconds: 5
+  periodSeconds: 5
+
+startupProbe:
+  httpGet:
+    path: /v1/health_check
+    port: nginx
+  initialDelaySeconds: 5
+  periodSeconds: 5
+  failureThreshold: 120
+
+# Override security context for gateway mode (simplified)
+securityContext: {}
```

Comment:
Why unset the security Context? We should use the default securityContext defined in values.yaml.